test/unit/controllers/CountrySelectModalControllerSpec.js

```diff
@@ -0,0 +1,149 @@
+      expect(this.$scope.slice).toEqual({limit: 20, limitDelta: 20})
+      done()
+    })
+
+    it('creates a sorted list of all selectable countries', function (done) {
+      this.$rootScope.$digest()
+      var expected = [{ name: 'Abkhazia', code: '+7 840' }, { name: 'Abkhazia', code: '+7 940' }, { name: 'Abkhazia', code: '+995 44' }, { name: 'Netherlands', code: '+31' }, { name: 'Vatican City', code: '+39 06 698' }, { name: 'Vatican City', code: '+379' }]
+
+      expect(this.$scope.countries).toEqual(expected)
+      done()
+    })
+
+    it('creates a sorted list of all selectable countries for an empty string-input', function (done) {
+      this.$rootScope.$digest()
+      this.$scope.search.query = ''
+      this.$rootScope.$digest()
+      var expected = [{ name: 'Abkhazia', code: '+7 840' }, { name: 'Abkhazia', code: '+7 940' }, { name: 'Abkhazia', code: '+995 44' }, { name: 'Netherlands', code: '+31' }, { name: 'Vatican City', code: '+39 06 698' }, { name: 'Vatican City', code: '+379' }]
+
+      expect(this.$scope.countries).toEqual(expected)
+      done()
+    })
+
+    describe(', when an input is given,', function () {
+      beforeEach(function () {
+        this.$rootScope.$digest()
+        this.$scope.search.query = 'A'
+      })
+
+      it('creates a sorted list of all countries containing the input', function (done) {
+        var expected = [{ name: 'Abkhazia', code: '+7 840' }, { name: 'Abkhazia', code: '+7 940' }, { name: 'Abkhazia', code: '+995 44' }, { name: 'Netherlands', code: '+31' }, { name: 'Vatican City', code: '+39 06 698' }, { name: 'Vatican City', code: '+379' }]
+
+        expect(this.$scope.countries).toEqual(expected)
+
+        this.$rootScope.$digest()
+        expected = [{ name: 'Abkhazia', code: '+7 840' }, { name: 'Abkhazia', code: '+7 940' }, { name: 'Abkhazia', code: '+995 44' }]
+
+        expect(this.$scope.countries).toEqual(expected)
+        done()
+      })
+
+      it('restore the original list when the input is deleted', function (done) {
+        this.$rootScope.$digest()
+        this.$scope.search.query = ''
+        this.$rootScope.$digest()
+
+        var expected = [{ name: 'Abkhazia', code: '+7 840' }, { name: 'Abkhazia', code: '+7 940' }, { name: 'Abkhazia', code: '+995 44' }, { name: 'Netherlands', code: '+31' }, { name: 'Vatican City', code: '+39 06 698' }, { name: 'Vatican City', code: '+379' }]
+
+        expect(this.$scope.countries).toEqual(expected)
+        done()
+      })
+
+      it('restore the original list when the input is deleted', function (done) {
+        this.$rootScope.$digest()
+        this.$scope.search.query = 'Ne'
+        this.$rootScope.$digest()
+
+        var expected = [{ name: 'Netherlands', code: '+31' }]
+
+        expect(this.$scope.countries).toEqual(expected)
+        done()
+      })
+    })
+
+    describe(', when no sorting is available,', function () {
+      beforeEach(function () {
+        String.prototype.localeCompare = null
```

Comment:
Ouch. Maybe store the original implementation in another variable and restore the prototype in `afterEach`. Otherwise you would have side effects if the order of test execution ever changes.